packages/venia-ui/lib/components/Routes/authRoute.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { bool, node, string } from 'prop-types';
+import { Redirect, Route, useLocation } from 'react-router-dom';
+
+import { useUserContext } from '@magento/peregrine/lib/context/user';
+
+const AuthRoute = props => {
+    const { authed, redirectTo, children, ...rest } = props;
+    const { pathname } = useLocation();
+    const [{ isSignedIn }] = useUserContext();
+
+    if (authed && !isSignedIn) {
+        return (
+            <Redirect
+                to={{
+                    pathname: redirectTo,
+                    // Keep path in memory to redirect user there when signed in
+                    state: { from: pathname }
+                }}
+            />
+        );
+    }
+
+    return <Route {...rest}>{children}</Route>;
+};
+
+AuthRoute.defaultProps = {
+    authed: false,
```

Comment:
I vaguely remember this being brought up, and can't remember why we went this direction. I would expect that any `<AuthRoute>` is assumed to need authentication and you wouldn't need this property.